Sources/Sentry/SentryRateLimitParser.m

```diff
@@ -12,41 +12,50 @@ @interface SentryRateLimitParser ()
 @implementation SentryRateLimitParser
 
 - (NSDictionary<NSNumber *, NSDate *> *)parse:(NSString *)header {
-    
-    NSMutableDictionary<NSNumber *, NSDate *> *rateLimits = [[NSMutableDictionary alloc] init];
-    
     if ([header length] == 0)  {
-        return rateLimits;
+        return @{};
     }
     
+    NSMutableDictionary<NSNumber *, NSDate *> *rateLimits = [NSMutableDictionary new];
+    
     // The header might contain whitespaces and they must be ignored.
     NSString *headerNoWhitespaces = [self removeAllWhitespaces:header];
     
-    NSArray<NSString *> *quotaLimits = [headerNoWhitespaces componentsSeparatedByString:@","];
-    
     // Each quotaLimit exists of retryAfter:categories:scope. The scope is ignored here
     // as it can be ignored by SDKs.
-    for (NSString* quota in quotaLimits) {
+    for (NSString* quota in [headerNoWhitespaces componentsSeparatedByString:@","]) {
         NSArray<NSString *> *parameters = [quota componentsSeparatedByString:@":"];
         
         NSNumber *retryAfterInSeconds = [self getRetryAfterInSeconds:parameters[0]];
         if (nil == retryAfterInSeconds || [retryAfterInSeconds intValue] <= 0) {
             continue;
         }
         
-        // The categories are a semicolon separated list. If this parameter is empty it stands
-        // for all categories. componentsSeparatedByString returns one category even if this
-        // parameter is empty.
-        NSArray<NSString *> *categories =  [parameters[1] componentsSeparatedByString:@";"];
-        for (NSString *categoryAsString in categories) {
-            SentryRateLimitCategory category = [self mapStringToCategory:categoryAsString];
-            rateLimits[[NSNumber numberWithInt:category]] = [SentryCurrentDate.date dateByAddingTimeInterval:[retryAfterInSeconds doubleValue]];
+        for (NSNumber *category in [self getCategories:parameters[1]]) {
+            rateLimits[category] = [SentryCurrentDate.date dateByAddingTimeInterval:[retryAfterInSeconds doubleValue]];
         }
     }
     
     return rateLimits;
 }
 
+- (NSArray<NSNumber *>*)getCategories:(NSString *)categoriesAsString {
```

Comment:
L: Can we call it `parse` instead of `get`?

```suggestion
- (NSArray<NSNumber *>*)parseCategories:(NSString *)categoriesAsString {
```

Reply:
Good point, I like it too.